cmd/podman/spec.go

```diff
@@ -550,6 +551,25 @@ func (c *createConfig) GetVolumeMounts() ([]spec.Mount, error) {
 			Options:     options,
 		})
 	}
+
+	// volumes from image config
+	for _, vol := range c.ImageVolumes {
```

Comment:
Can't we merge this and the AddImageVolumes section  you have below.  rather then relying on a Souce=="" flag, just create the content on the fly.

Reply:
Also, we should add a check here to make sure we don't override user-set volumes

Reply:
Added check, fixed.